setup.py - cython version check: ValueError: invalid literal for int() with base 10: '1post0'

Should be more failsave I think...

```
for PYX in python2.7 python3.4 ; do \
	    /usr/bin/make build PYTHON=$PYX; \
	done
make[2]: Entering directory `/build/buildd/kivy-1.9.0-dev~testing0+201502071058-2609'
python2.7 setup.py build_ext --inplace
Traceback (most recent call last):
  File "setup.py", line 115, in <module>
    cy_version = tuple(map(int, cy_version_str.split('.')))
ValueError: invalid literal for int() with base 10: '1post0'
make[2]: *** [build] Error 1
make[2]: Leaving directory `/build/buildd/kivy-1.9.0-dev~testing0+201502071058-2609'
make[2]: Entering directory `/build/buildd/kivy-1.9.0-dev~testing0+201502071058-2609'
python3.4 setup.py build_ext --inplace
Traceback (most recent call last):
  File "setup.py", line 115, in <module>
    cy_version = tuple(map(int, cy_version_str.split('.')))
ValueError: invalid literal for int() with base 10: '1post0'
```
- https://launchpadlibrarian.net/196899423/buildlog_ubuntu-trusty-amd64.kivy_1.9.0-dev~testing0%2B201502071058-2609-testing65~ubuntu14.04.1_FAILEDTOBUILD.txt.gz

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/8371553-setup-py-cython-version-check-valueerror-invalid-literal-for-int-with-base-10-1post0?utm_campaign=plugin&utm_content=tracker%2F42681&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F42681&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/setup.py
+++ b/setup.py
@@ -3,11 +3,6 @@
 # http://kivy.org/
 #
 
-MIN_CYTHON_VERSION = (0, 20, 0)
-MIN_CYTHON_STRING = '.'.join(map(str, MIN_CYTHON_VERSION))
-MAX_CYTHON_VERSION = (0, 21, 1)
-MAX_CYTHON_STRING = '.'.join(map(str, MAX_CYTHON_VERSION))
-
 import sys
 
 from copy import deepcopy
@@ -16,7 +11,14 @@ from os.path import join, dirname, sep, exists, basename
 from os import walk, environ
 from distutils.core import setup
 from distutils.extension import Extension
+from distutils.version import LooseVersion
 from collections import OrderedDict
+
+
+MIN_CYTHON_STRING = '0.20.0'
+MIN_CYTHON_VERSION = LooseVersion(MIN_CYTHON_STRING)
+MAX_CYTHON_STRING = '0.21.1'
+MAX_CYTHON_VERSION = LooseVersion(MAX_CYTHON_STRING)
 
 if sys.version > '3':
     PY3 = True
@@ -112,13 +114,13 @@ else:
         have_cython = True
         import Cython
         cy_version_str = Cython.__version__
-        cy_version = tuple(map(int, cy_version_str.split('.')))
+        cy_ver = LooseVersion(cy_version_str)
         print('\nDetected Cython version {}'.format(cy_version_str))
-        if cy_version < MIN_CYTHON_VERSION:
+        if cy_ver < MIN_CYTHON_VERSION:
             print('  This version of Cython is not compatible with Kivy. ' +
                   'Please upgrade to at least {}'.format(MIN_CYTHON_STRING))
             raise ImportError('Incompatible Cython Version')
-        if cy_version > MAX_CYTHON_VERSION:
+        if cy_ver > MAX_CYTHON_VERSION:
             print('  This version of Cython is untested with Kivy. If you ' +
                   'experience issues, please downgrade to {}'
                   .format(MAX_CYTHON_STRING))

